### PR TITLE
#1933 Consider partially connected elements in lost equipment report

### LINKED ISSUE #1933: Fix computation of lost equipment
Equipment completely connected at the initial state and then only partially disconnected at the final state should NOT be considered as lost. Also, a partially connected equipment at the initial state and totally disconnected at the final state should be considered as lost.

Currently, to determine the list of lost equipment, the `isConnected` method in the data interface is used to check initial and final state of an equipment.

The interpretation of `isConnected` for equipment with more than one end states that it is connected only if all the ends are connected. This interpretation must not be changed. With this interpretation for `isConnected`, lines that are initially connected at both ends and are left opened in the final state only at one end are considered as lost. This is considered wrong.

To fix the current implementation of the computation of lost equipment, a new API `isPartiallyConnected` will be proposed.

### PR DESCRIPTION
**Pull request by zamarrenolm**
Closes #1933

### CODE REVIEW

**Comment by zamarrenolm** [comment]
Changes squashed and branch rebased on master.

File: dynawo/sources/Modeler/DataInterface/IIDM/test/TestThreeWTransformerInterface.cpp [hunk 1]
```diff
@@ -0,0 +1,158 @@
+//
+// Copyright (c) 2015-2020, RTE (http://www.rte-france.com)
```
Comment (gautierbureau) [inline dynawo/sources/Modeler/DataInterface/IIDM/test/TestThreeWTransformerInterface.cpp:2]: 2021

**Review by gautierbureau — APPROVED** [review]

File: dynawo/sources/Modeler/DataInterface/DYNComponentInterface.h [hunk 2]
```diff
@@ -136,6 +136,12 @@ class ComponentInterface {
    */
   virtual bool isConnected() const;
 
+  /**
+   * @brief check the connection state of the component
+   * @return @b true is the component is connected at least at one end, @b false else
```
Comment (marcochiaramello) [inline dynawo/sources/Modeler/DataInterface/DYNComponentInterface.h:141]: typo, true if

File: dynawo/sources/Modeler/DataInterface/IIDM/test/TestThreeWTransformerInterface.cpp (hunk 1 above)
Comment (zamarrenolm) [reply dynawo/sources/Modeler/DataInterface/IIDM/test/TestThreeWTransformerInterface.cpp]: Fixed year

File: dynawo/sources/Modeler/DataInterface/DYNComponentInterface.h (hunk 2 above)
Comment (zamarrenolm) [reply dynawo/sources/Modeler/DataInterface/DYNComponentInterface.h]: Fixed comment in `isConnected` and `isPartiallyConnected` methods.

**Review by marcochiaramello — APPROVED** [review]

**Pull request merged**